src/components/VideoAsset.vue

```diff
@@ -1,33 +1,41 @@
 <!--
   This source file is part of the Swift.org open source project
 
-  Copyright (c) 2021 Apple Inc. and the Swift project authors
+  Copyright (c) 2021-2023 Apple Inc. and the Swift project authors
   Licensed under Apache License v2.0 with Runtime Library Exception
 
   See https://swift.org/LICENSE.txt for license information
   See https://swift.org/CONTRIBUTORS.txt for Swift project authors
 -->
 
 <template>
-  <video
-    :controls="showsControls"
-    :autoplay="autoplays"
-    :poster="normalisedPosterPath"
-    :muted="muted"
-    :width="optimalWidth"
-    playsinline
-    @playing="$emit('playing')"
-    @pause="$emit('pause')"
-    @ended="$emit('ended')"
+  <ConditionalWrapper
```

Comment:
Could you briefly explain the need for `<ConditionalWrapper>`? I think you added it because you want to conditionally wrap the video in another element only if it has a device frame, right?

And you didn't just use normal `v-if` templating because it would make the inner `<video ...>` part have to be copied twice?

And `<component :is="...">` would be limited to using a plain div or something in the case where there is no device frame?

Is that all right? Just making sure I understand the need for it.

Reply:
Yes exactly all of those. I wanted to make it as transparent as possible. 

Initially I did it by moving everything to render functions in the VideoAsset. By doing that, you can easily wrap things, but I then reverted back to the template syntax since it read easier, and we may use this handy component somewhere else in the future?